How to Upgrade to v2 (#500)
vshn/k8up: docs/modules/ROOT/pages/how-tos/upgrade.adoc

= How to Upgrade K8up

WARNING: Never remove the CRDs, as you might lose the respective resources!

[#upgrade_1_to_2]
== Upgrade K8up from 1.x to 2.x

TIP: See also xref:explanations/what-has-changed-in-v2.adoc[].

> Only little has changed on from a user's perspective in K8up 2.x, but a lot did change under the hood nonetheless.

Reviewer: ```suggestion
Because of the move of K8up to it's own GitHub organization, and a cleanup of several namings, there are visible changes from a user's perspective in K8up 2.x, which require (manual) intervention for a successful upgrade.
```

> Some of these changes may require you to update your configuration of K8up.

Reviewer 2: With the CRD rename AND version bump from v1alpha1 to v1 in #509 , I'm not sure we can call it "only little has changed"

> Some of these changes may require you to update your configuration of K8up.

Reviewer: ```suggestion
These changes require you to update your configuration of K8up.
```

- In the CRD definition, what was `backup.appuio.ch` is now `k8up.io`.
  We've also updated the version from `v1alpha` to `v1`.
  So, for example, `backups.backup.appuio.ch/v1alpha1` becomes `backups.k8up.io/v1`.
- The default values for the annotations have been changed from `k8up.syn.tools/*` to `k8up.io/*`.
  Internally used annotations will be handled transparently.
  But annotations, which have been added to resources (such as `k8up.syn.tools/backupcommand`) need to be updated accordingly (in the previous example to `k8up.io/backupcommand`).
  Alternatively, you can also change the annotations back in xref:references/operator-config-reference.adoc[the operator's configuration].
  In the previous example, you could set `BACKUP_BACKUPCOMMANDANNOTATION` to `k8up.syn.tool/backupcommand` to overwrite the new default.
- If you directly reference our Go code in your Go code, then you will need to update the imports.
  The k8up Go module is now called https://github.com/k8up-io/k8up[`github.com/k8up-io/k8up`].

Our docs are still at https://k8up.io.
Our reference helm chart is still at https://github.com/appuio/charts/ and was updated accordingly.

=== Wrestic Integration

> We have integrated the code previously known as the `wrestic` into k8up.

Reviewer: ```suggestion
We have integrated the code previously known as `wrestic` into k8up.
```
